uni/lib/controller/parsers/parser_exams.dart

```diff
@@ -37,27 +38,30 @@ class ParserExams {
         .querySelectorAll('div > table > tbody > tr > td')
         .forEach((Element element) {
       element.querySelectorAll('table:not(.mapa)').forEach((Element table) {
-        table.querySelectorAll('th').forEach((Element week) {
-          weekDays.add(week.text.substring(0, week.text.indexOf('2')));
-        });
         table.querySelectorAll('span.exame-data').forEach((Element date) {
           dates.add(date.text);
         });
-
         table.querySelectorAll('td.l.k').forEach((Element exams) {
           if (exams.querySelector('td.exame') != null) {
             exams.querySelectorAll('td.exame').forEach((Element examsDay) {
               if (examsDay.querySelector('a') != null) {
-                subject = examsDay.querySelector('a')?.text;
+                subject = examsDay.querySelector('a')!.text;
+                id = examsDay.querySelector('a')!.attributes['href']!.split('=')[1];
```

Comment:
Good approach but you are working with an url, why not use [Uri.parse](https://api.dart.dev/stable/2.18.4/dart-core/Uri/parse.html) and [get the query parameter p_exa_id](https://api.dart.dev/stable/2.18.4/dart-core/Uri/queryParameters.html)?

It's a safer solution

Reply:
I'll explore this approach.